scanamo/src/main/scala/org/scanamo/SecondaryIndex.scala

```diff
@@ -43,6 +44,25 @@ sealed abstract class SecondaryIndex[V] {
     */
   def scan(): ScanamoOps[List[Either[DynamoReadError, V]]]
 
+  /**
+    * Performs a scan with the ability to introduce effects into the computation. This is
+    * useful for huge tables when you don't want to load the whole of it in memory, but
+    * scan it page by page.
+    *
+    * To control how many maximum items to load at once, use [[scanToPaged]]
+    */
+  final def scanTo[M[_]: Alternative]: ScanamoOpsT[M, List[Either[DynamoReadError, V]]] = scanToPaged(Int.MaxValue)
+
+  /**
+    * Performs a scan with the ability to introduce effects into the computation. This is
+    * useful for huge tables when you don't want to load the whole of it in memory, but
+    * scan it page by page, with a maximum of `pageSize` items per page..
+    *
+    * @note DynamoDB will only ever return maximum 1MB of data per scan, so `pageSize` is an
+    * upper bound.
+    */
+  def scanToPaged[M[_]: Alternative](pageSize: Int): ScanamoOpsT[M, List[Either[DynamoReadError, V]]]
+
```

Comment:
Would `paginated` better than `paged`? I am not a native english speaker so feel free to ignore my comment.